### ISSUE TITLE
Socket based auth doesn't use passwords

### ISSUE THREAD

**Issue by OlafvdSpek** [phpmyadmin/phpmyadmin#14719]
But the new user form requires me to enter a password and the generated query does seem to include that password. Not sure what MariaDB does with the password, it didn't return an error.

**Comment by ibennetch** [comment]
I'm a little confused. When I use the MySQL command line client to connect via socket connection, I need to specify a username and password. PHP/phpMyAdmin is the same way; you need to provide valid credentials. Are you talking about the [`auth_socket` pluggable authentication method](https://dev.mysql.com/doc/refman/5.7/en/socket-pluggable-authentication.html#socket-pluggable-authentication-usage)?

**Comment by OlafvdSpek** [comment]
Yes I am.

**Comment by williamdes** [comment]
Config auth does not use username and password.
Username and password should be hidden or disabled.
Use a multi server setup that has a config setup and a cookie setup.

**Comment by ibennetch** [comment]
I don't think we support auth_socket at all right now, this probably should become a request to add functionality for that in more places than just the new user area.

**Comment by OlafvdSpek** [comment]
This one is not about using socket auth for pMA logins..

**Comment by williamdes** [comment]
https://mariadb.com/kb/en/library/authentication-plugin-unix-socket/
```sql
INSTALL PLUGIN unix_socket SONAME 'auth_socket';
CREATE USER 'william'@'localhost' IDENTIFIED WITH unix_socket AS 'william';
```

```php
$i++;
$cfg['Servers'][$i]['auth_type'] = 'config';
$cfg['Servers'][$i]['socket'] = '/var/run/mysqld/mysqld.sock';
$cfg['Servers'][$i]['user'] = 'william';
$cfg['Servers'][$i]['AllowNoPassword'] = true;
```

That config only works because `william` is my Ubuntu username.

Required:
```php
$cfg['Servers'][$i]['AllowNoPassword'] = true;
```

**Comment by williamdes** [comment]
@OlafvdSpek What is the issue ?
Username and password inputs not used ?

**Comment by OlafvdSpek** [comment]
> But the new user form requires me to enter a password and the generated query does seem to include that password.

;)

**Comment by williamdes** [comment]
@OlafvdSpek `php7.2-fpm` is running as `william:www-data`

**Comment by OlafvdSpek** [comment]
Does it allow logins with any password or just with the empty password?
And with any username or just when you enter `william` in the pMA login form?

**Comment by apoorvkhare07** [comment]
@williamdes What exactly is the issue here ? I pretty much got what @OlafvdSpek is saying. But what behaviour do you expect ?

**Comment by williamdes** [comment]
@apoorvkhare07 I think we should disable inputs (user, password) for socket and config auth options in the UI

**Comment by ibennetch** [comment]
I’m not sure how to handle this, because it doesn’t really fit in tomour
auth_type model (since it’s configurable per-user). It doesn’t really fit
in to our existing login models because you need to specify the user name
(which normally would come from the Unix username, but in our case that’s
going to be the web server process.

The best idea I have right now is to implement it as a new auth_type,
operating similar to cookie. Olaf, do you have a vision for what logging in
this way would look like?

**Comment by OlafvdSpek** [comment]
@ibennetch 
There are multiple aspects, one is editing users and two is signing into pMA itself.
This ticket is about the former.

Not sure why @williamdes wants to disable the username input, just disabling the password input seems like the right thing to do.

**Comment by williamdes** [comment]
@OlafvdSpek I want to disable input for 'config' authentication method because it does not make sense

**Comment by OlafvdSpek** [comment]
What form / screen are you talking about?

**Comment by williamdes** [comment]
@OlafvdSpek This one, the login screen
![image](https://user-images.githubusercontent.com/7784660/55666403-db36b600-584e-11e9-8a62-0095690cab38.png)

**Comment by OlafvdSpek** [comment]
This issue is about this form:

![pma](https://user-images.githubusercontent.com/2218936/55666489-edfdba80-584f-11e9-832c-ab73fd0cb6f7.PNG)

**Comment by williamdes** [comment]
Okayyy, sorry for the confusion
